spec/requests/search_spec.rb

```diff
@@ -1,64 +1,70 @@
 # frozen_string_literal: true
-describe SearchController, type: :controller do
+
+describe 'Search', type: :request do
   let(:user) { create(:user) }
-  let(:user2) {  create(:user, email: 'bar@email.com') }
-  let(:user3) {  create(:user, email: 'foo@email.com', banned: true) }
+  let!(:user2) do
+    create(:user, email: 'bar@email.com',
+                  name: 'user 2 name')
+  end
+  let!(:user3) do
+    create(:user, email: 'foo@email.com',
+                  banned: true,
+                  name: 'user 3 name')
+  end
 
   describe '#index' do
     context 'when user is logged in' do
-      include_context :logged_in_user
+      before { sign_in user }
 
       context 'when have passed email' do
         it 'strip passed string' do
-          expect_any_instance_of(String).to receive(:strip)
-          get :index, params: { search: { email: '  hi@email.com ' } }
+          get search_index_path, params: { search: { email: 'bar@email.com' } }
           expect(response).to render_template(:index)
         end
 
         it 'filters by non-banned user email' do
-          get :index, params: { search: { email: 'bar@email.com' } }
-          expect(assigns(:matching_users)).to include(user2)
+          get search_index_path, params: { search: { email: 'bar@email.com' } }
           expect(response).to render_template(:index)
+          expect(response.body).to include(user2.name)
         end
 
         it 'does not filter banned user email' do
-          get :index, params: { search: { email: 'foo@email.com' } }
-          expect(assigns(:matching_users)).not_to include(user3)
+          get search_index_path, params: { search: { email: 'foo@email.com' } }
+          expect(response.body).not_to include(user3.name)
           expect(response).to render_template(:index)
         end
 
         it 'keeps a reference to the email queried' do
-          get :index, params: { search: { email: 'bar@email.com' } }
-          expect(assigns(:email_query)).to include('bar@email.com')
+          get search_index_path, params: { search: { email: 'bar@email.com' } }
+          expect(response.body).to include(CGI.escape('bar@email.com'))
```

Comment:
Why is `CGI.escape()` necessary?

Reply:
In the `response.body` the html response have email encoded format `bar%40email.com` instead of `bar@email.com`. Do you know another to do this assertion?

Reply:
I tried just `expect(response.body).to include('bar@email.com')` and it works for me. 

When you type in the search box and it says `User bar@email.com not found`. You're just asserting the `response.body` that the following text is there which is in a string format. 

Here's a screenshot as reference for what we would be trying to assert.

![image](https://user-images.githubusercontent.com/7613125/95020152-5af93880-061e-11eb-85ea-3c8dcebe1b5a.png)

Reply:
I think if you do `JSON.parse(response.body)` it will use the actual `@` character. Let's use this instead since we do this in other similar tests.

Reply:
@julianguyen Interesting. `JSON.parse` actually fails for me while `response.body` itself works. It seems like using `JSON.parse(response.body)` throws an exception for `JSON::ParserError` `784: unexpected token at '<!DOCTYPE html>`.

`CGI.escape` wasn't needed when I tried testing it.

Reply:
Thanks for verifying! :) +1 to removing. `CGI.escape` then!

Reply:
yeah. It worked without `CGI.escape` 
sorry last time I tried I got `bar%40email.com` that's why I had added this logic.